checkstyle/checkstyle.xml

```diff
@@ -98,7 +98,7 @@
     <module name="MethodLength"/>
     <module name="ParameterNumber">
       <!-- default is 8 -->
-      <property name="max" value="12"/>
+      <property name="max" value="13"/>
```

Comment:
I think there is still some value in keeping this check. The reason the number is being bumped here is because the Sender constructor has added an argument. The solution would be to have a builder, but then a builder doesn't make sense for the Sender. 

It may make sense to just exempt the `Sender` for this check, but I am not sure if that is possible.